identity/ideth/factory.go

```diff
@@ -123,13 +125,29 @@ func (s *factory) CreateIdentity(ctx context.Context) (did *id.DID, err error) {
 	// non async task
 	if !isDone {
 		return nil, errors.New("Create Identity TX failed: txID:%s", txID.String())
+	}
+
+	tx, err := s.txManager.GetTransaction(createdDID, txID)
+	if err != nil {
+		return nil, err
+	}
+	idCreated, ok := tx.Values[identityCreatedEventName]
+	if !ok {
+		return nil, errors.New("Couldn't find value for %s", identityCreatedEventName)
+	}
+	createdAddr := common.BytesToAddress(idCreated.Value)
+	log.Infof("ID Created with address: %s", createdAddr.Hex())
 
+	if calcIdentityAddress.Hex() != createdAddr.Hex() {
+		log.Infof("[Recovered] Found race condition creating identity, calculatedDID[%s] vs createdDID[%s]", calcIdentityAddress.Hex(), createdAddr.Hex())
 	}
 
-	err = isIdentityContract(*identityAddress, s.client)
+	err = isIdentityContract(createdAddr, s.client)
```

Comment:
We need to update the isIdentityContract method.

Instead of comparing the bytes we should ask the identityFactory if it created the identity.